app/2.0/docs/about_20.md

```diff
@@ -0,0 +1,438 @@
+
+The legacy <code>created</code> callback is no longer called before default values in <code>properties</code> have been set.  As such, you should not rely on properties set in <code>created</code> from within <code>value</code> functions that define property defaults.
+<p>
+However, you can now set <strong>any</strong> property defaults within the <code>created</code> callback (in 1.0 this was forbidden for observed properties) in lieu of using the <code>value</code> function in <code>properties</code>.
+   </td>
+  </tr>
+  <tr>
+   <td><code>connectedCallback (</code>native<code>)</code>
+<p>
+<code>attached</code> (legacy)
+   </td>
+   <td>Polymer 1.x deferred the <code>attached</code> callback until after first render, so elements could measure themselves or their children.
+   </td>
+  </tr>
+  <tr>
+   <td><code>disconnectedCallback (</code>native<code>)</code>
+<p>
+<code>detached</code> (legacy)
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td><code>attributeChangedCallback (</code>native<code>)</code>
+<p>
+<code>attributeChanged</code> (legacy)
+   </td>
+   <td>Attributes must be <em>explicitly</em> registered to be observed.
+<p>
+For Polymer elements, only properties explicitly declared in the  <code>properties</code> object are tracked for attribute changes. (That is, changing the attribute value invokes the attribute changed callback, and causes Polymer to set the property value from the attribute.)
+<p>
+In custom elements v0, the <code>attributeChangedCallback</code> was generated for <strong>any</strong> attribute change.
+<p>
+In Polymer 1.x, attributes were deserialized for both explicitly declared properties and  <em>implicitly declared properties</em>. For example, a property used in a binding or as the dependency of an observer, but not declared in <code>properties</code> can be considered implicitly declared.
+   </td>
+  </tr>
+  <tr>
+   <td><code>ready</code> (Polymer specific)
+   </td>
+   <td>Polymer longer guarantees that the initial light DOM distribution is complete before ready is called.
+   </td>
+  </tr>
+</table>
+
+
+In addition to changes in the callbacks, note that the `lazyRegister` option has been removed and all meta-programming (parsing the template, creating accessors on the prototype, and so on) is deferred until the first instance of the element is created.
+
+### Type-extension elements {#type-extension}
+
+Polymer 2.0 doesn't support type-extension elements (for example, `<input is="iron-input">`). Type-extension support is still included in the custom elements v1 spec (as "customized build-in elements"), and scheduled for implementation in Chrome. However, since Apple has said it will not implement `is`, we will not be encouraging its use to avoid indefinite reliance on the custom elements polyfill. Instead, a wrapper custom element can surround a native element. For example:
+
+`<a is="my-anchor">...</a>`
+
+Could become:
+
+
+```
+<my-anchor>
+  <a>...</a>
+</my-anchor>
+```
+
```

Comment:
I think the answer to #2 is "yes" (we're backing off from pushing type-extension elements in the short term). I don't think it has anything to do with progressive enhancement. 

I guess the text here could be a little more nuanced. For something like `<dom-bind>`, the whole point is taking a user-supplied template... Hence, taking a template supplied in light DOM makes sense.

A custom input _could_ just as well use an input element in shadow DOM (which is what `<paper-input>` used to do, and probably still does). However, there are a *lot* of attributes you can tweak on an input element... So rather than trying to proxy all of those through to your custom element API, it's easier to let the user set them directly, which the old `iron-input` did by virtue of being a type-extension element, and the new `iron-input` does by taking an `<input>` in light DOM.